src/elements/paginator/paginator.ts

```diff
@@ -0,0 +1,377 @@
+    return this._pageSizeOptions;
+  }
+  private _pageSizeOptions?: number[];
+
+  /**
+   * Position of the prev/next buttons: if `pageSizeOptions` is set, the sbb-select for the pageSize change
+   * will be positioned oppositely with the page numbers always in the center.
+   */
+  @property({ attribute: 'pager-position', reflect: true }) public pagerPosition: 'start' | 'end' =
+    'start';
+
+  /** Size variant, either m or s. */
+  @property({ reflect: true }) public size: 'm' | 's' = 'm';
+
+  private _page: EventEmitter<SbbPaginatorPageEventDetails> = new EventEmitter(
+    this,
+    SbbPaginatorElement.events.page,
+    { composed: true, bubbles: true },
+  );
+
+  private _paginatorOptionsLabel = `sbb-paginator-options-label-${++optionsLabelNextId}`;
+  private _language = new SbbLanguageController(this);
+  private _markForFocus: number | null = null;
+  private _updateSelectAriaLabelledBy: boolean = false;
+
+  protected override updated(changedProperties: PropertyValues<this>): void {
+    super.updated(changedProperties);
+
+    /** Tab navigation can force a rerender when ellipsis elements need to be displayed; the focus must stay on the correct element. */
+    if (this._markForFocus && sbbInputModalityDetector.mostRecentModality === 'keyboard') {
+      const focusElement = this._getVisiblePages().find(
+        (e) => this.pageIndex === +e.getAttribute('data-index')!,
+      );
+      if (focusElement) {
+        (focusElement as HTMLElement).focus();
+      }
+      // Reset mark for focus
+      this._markForFocus = null;
+    }
+
+    /**
+     * TODO: Accessibility fix required to correctly read the label;
+     * can be possibly removed after the merge of https://github.com/sbb-design-systems/lyne-components/issues/3062
+     */
+    const select = this.shadowRoot!.querySelector('sbb-select');
+    if (select && this._updateSelectAriaLabelledBy) {
+      select.setAttribute('aria-labelledby', this._paginatorOptionsLabel);
+      this._updateSelectAriaLabelledBy = false;
+    }
+
+    // To reliably announce page change, we have to set the label in updated() (a tick later than the other changes).
+    this.shadowRoot!.querySelector('sbb-screen-reader-only')!.textContent =
+      this._currentPageLabel();
+  }
+
+  /**
+   * Calculates the current number of pages based on the `length` and the `pageSize`;
+   * value must be rounded up (e.g. `length = 21` and `pageSize = 10` means 3 pages).
+   */
+  private _numberOfPages(): number {
+    if (!this.pageSize) {
+      return 0;
+    }
+    return Math.ceil(this.length / this.pageSize);
+  }
+
+  /**
+   * If the `pageSize` changes due to user interaction with the `pageSizeOptions` select,
+   * emit the `page` event and then update the `pageSize` value.
+   */
+  private _pageSizeChanged(value: number): void {
+    const previousPageSize = this._pageSize;
+    const newPageSize = Math.max(value, 0);
+    if (previousPageSize !== newPageSize) {
+      this._page.emit({
```

Comment:
The order could be a problem. We emit the event, which is synchronous and defines page index and page size, which is not yet updated?